### PR TITLE
Lecture 22 Network Centrality

### PR DESCRIPTION
**Pull request by sladenheim**
Summary of the PR

- Format slides
- I elaborated more on the definitions of 
- Revised slides on eigenvector centrality
- Add colorbars and legends to all of the network plots
- Edited adjacency graph to show 1's and 0's

Addresses #157

### CODE REVIEW

**Review by trgardos — CHANGES_REQUESTED** [review]
oof. this is a pretty heavy lecture. I'm inclined to not go into all the details and derviations and maybe make up our time to fit it in the time remaining wed/thurs. what do you think?

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd [hunk 1]
```diff
@@ -79,160 +96,191 @@ ax2 = fig.add_subplot(122)
...
 
-We will study three basic notions of centrality:
-    
-1. __Closeness__ Centrality:  A central node is close to all others.
-2. __Betweenness__ Centrality: A central node is on many paths through the network.
-3. __Status__ Centrality: A central node is connected to other central nodes.
-
-### Closeness Centrality
+$$
+\text{closeness}(i) = \frac{1}{\sum_{j \in V} d(i,j)}.
```
[the remark this answers is not in the document]
Comment (trgardos) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:112]: I suppose it is the set of all vertices, right?

[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:112]: Yes.

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd [hunk 2]
```diff
@@ -246,66 +294,58 @@ def show_complete_array():
...
 
-In this context, one often talks of "prestige" rather than "centrality."  But the concepts are related.
+Given a graph with $n$ nodes, let $A$ be the adjacency matrix of the graph, where $A_{ij}$ is 1 if there is an edge from node $i$ to node $j$, and 0 otherwise.
 
-The idea of status centrality is that __"high status" nodes are those that are connected to "high status" nodes.__
+The eigenvector centrality $x_i$ of node $i$ is defined as
 
-If you think this definition is circular, you are right!
+$$
+x_i = \frac{1}{\lambda} \sum_{j=1}^{n} A_{ij} x_j,
```
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:318]: Fine with that.

**Comment by sladenheim** [comment]
I am fine with covering a few of the topics (maybe only 1 of the centrality measures) as opposed to all of it. I think we would only get through a small portion of this lecture.

**Comment by trgardos** [comment]
indeed, we could put conditional display around the derivations and leave that in web course notes and then just show results and examples in the slides.

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd [hunk 3]
```diff
@@ -246,66 +294,58 @@ def show_complete_array():
...
 
-So we can write even more concisely:
-    
-$$ A\mathbf{e} = \lambda\mathbf{e} $$
+$$
+\mathbf{Ax} = \lambda \mathbf{x},
+$$
 
-... which means that $\mathbf{e}$ must be an eigenvector of $A$.
+where $\mathbf{x}$ is the eigenvector corresponding to the the eigenvalue $\lambda$ of the adjacency matrix $\mathbf{A}$. The eigenvector $\mathbf{x}$ gives the centrality scores for all nodes in the network.
```
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:335]: Under the condition of a graph being strongly connected (the adjacency matrix is then irreducible), the Perron Frobenius theorem will imply that the eigenvector is unique https://nhigham.com/2021/07/13/what-is-the-perron-frobenius-theorem/. 

I agree we can make this less "theoretical".

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd (hunk 1 above)
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:112]: I added a line saying $G=(V, E)$ to make it clearer that $V$ is the set of nodes.

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd (hunk 2 above)
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:322]: I added a line about how $x_i$ is proportional to a sum of the nodes that it is actually connected to. Let me know if that works for you.

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd [hunk 4]
```diff
@@ -323,47 +363,61 @@ nx.draw_networkx(Gk, ax = ax2, pos = nx.spring_layout(Gk),
...
-plt.title('Distribution of Eigenvector Centrality', size = 16);
+plt.title('Distribution of Eigenvector Centrality', size = 16)
+plt.show()
 ```
 
-As a more detailed example, we can ask whether this definition of "prestige" applies to NCAA Division 1A football teams, based on who they play against.
+## Eigenvector Centrality: Football Dataset
 
-(Remember this data is from 2000!)
+As a more detailed example, we can ask whether this definition of *prestige* applies to NCAA Division 1A football teams, based on who they play against. Remember this data is from 2000.
```
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:391]: I commented out this example. It feels randomly thrown in and I feel it distracts from the point we are trying to make about the karate example.

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd [hunk 5]
```diff
@@ -323,47 +363,61 @@ nx.draw_networkx(Gk, ax = ax2, pos = nx.spring_layout(Gk),
...
+plt.axis('off')
+
+# Show the plot
+plt.show()
 ```
 
-So let's compare the three versions of centrality we've looked at:
+## Centrality Comparison: Karate Example
+
+Let's compare the three versions of centrality we've looked at so far.
```
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:432]: I made a comment saying we have highlighted 2 important individuals in the plot who are the teacher and club president.

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd [hunk 6]
```diff
@@ -383,75 +437,62 @@ for i in range(3):
...
-We would like to __isolate__ $s$ from $t$.  What is the cheapest way to do it?
+The goal of a min $s$-$t$ cut is to partition the graph into two disjoint subsets such that (s) is in one subset and (t) is in the other, and the total weight (or number) of the edges that need to be removed to achieve this separation is minimized.
 
 <!-- Image credit: On the history of the transportation and maximum flow problems. -->
 <!--     Alexander Schrijver in Math Programming, 91: 3, 2002. -->
     
-<center>
-    
-<img src="figs/L23-max-flow-soviet-rail-1955.png" width="75%">
+![](figs/L23-max-flow-soviet-rail-1955.png){fig-align="center" width="75%"}
```
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:488]: I added more context about what this figure is. The US government was looking where they could effectively halt transportation of soviet materials!

**Comment by sladenheim** [comment]
I made some revisions to the request. I think we should aim to just cover the centrality measures. We can then tell them to read about min-cut and balanced cuts on their own.

File: ds701_book/22-Networks-II-Centrality-Clustering.qmd (hunk 3 above)
[the remark this answers is not in the document]
Comment (sladenheim) [reply ds701_book/22-Networks-II-Centrality-Clustering.qmd:341]: I added an additional slide for the geometric interpretation of eigenvector centrality.

**Review by trgardos — APPROVED** [review]

**Pull request merged**